src/main/java/org/toradocu/extractor/JavadocExtractor.java

```diff
@@ -1,515 +1,664 @@
-    paramTags.addAll(paramTagsMap.values());
-    List<ParamTag> memberParamTags = new ArrayList<>();
-    for (Tag tag : paramTags) { // For each of the tags in paramTags.
-      if (!(tag instanceof com.sun.javadoc.ParamTag)) { // If it is not an instanceof ParamTag.
-        throw new IllegalStateException(
-            tag
-                + " is not a @param tag. This should not happen."
-                + " Toradocu only considers @param and @throws tags.");
-      }
+  private String removePackage(String type, boolean isConstructor) {
+    // Constructor names contain package name.
+    int lastDot = type.lastIndexOf(".");
+    if (lastDot != -1) {
+      type = type.substring(lastDot + 1);
+    }
+
+    // Constructor of nested classes need a special parsing
+    int dollar = type.indexOf("$");
+    if (isConstructor && dollar != -1) {
+      type = type.substring(dollar + 1, type.length());
+    }
+
+    return type;
+  }
 
-      // We create a paramsTag that will be the cast of tag to ParamTag in order to work with it.
-      com.sun.javadoc.ParamTag paramsTag = (com.sun.javadoc.ParamTag) tag;
-
-      // Handle inline taglets such as {@inheritDoc}.
-      TagletWriter tagletWriter =
-          new HtmlDocletWriter(configuration, DocPath.forClass(classDoc))
-              .getTagletWriterInstance(false);
-      String comment = tagletWriter.commentTagsToOutput(tag, tag.inlineTags()).toString();
-
-      // Remove HTML tags (also generated by inline taglets). In the future, perhaps retain those
-      // tags, because they contain information that can be exploited.
-      comment = Jsoup.parse(comment).text();
-
-      // The list of the parameters of the method that we'll use in order to match
-      // the parameterName of the tag with the parameter itself
-      List<Parameter> parameters = getParameters(member);
-
-      String name = null; //Name of the ParamTag that we'll introduce
-      org.toradocu.extractor.Type type = null; //type of the parameter of the ParamTag
-      Boolean nullable = null; // Nullability of the parameter.
-
-      boolean found = false; //Boolean for control
-      for (int i = 0; !found && i < parameters.size(); i++) {
-        if (parameters.get(i).getName().equals(paramsTag.parameterName())) {
-          // If the tag parameterName matches any parameter name in the method, then we stop the
-          // loop.
-          found = true;
-          name = parameters.get(i).getName(); //then we assign values to the variables
-          type = parameters.get(i).getType();
-          nullable = parameters.get(i).getNullability();
+  /**
+   * Search for the type of the exception with the given type name. The type name is allowed to be
+   * fully-qualified or simple, in which case this method tries to guess the package name.
+   *
+   * @param classesInPackage list of class names in sourceMember's package
+   * @param sourceMember the source member for which the exception with type name {@code
```

Comment:
As elsewhere, use the more specific "callable" or "executable" rather than "member".